### ISSUE TITLE
Removing BMP/modification causes graphs to vanish

### ISSUE THREAD

**Issue by jamesmcclain** [WikiWatershed/model-my-watershed#921]
On the modeling page, removing a BMP or reclassification causes the graphs to disappear:

No reclassification initially:
![screenshot from 2015-10-07 17 10 54](https://cloud.githubusercontent.com/assets/11281373/10351414/6f5c939e-6d17-11e5-82c4-b6a0538a8abb.png)

Reclassification added:
![screenshot from 2015-10-07 17 11 21](https://cloud.githubusercontent.com/assets/11281373/10351429/843d65f4-6d17-11e5-8646-14819c401412.png)

Reclassification removed:
![screenshot from 2015-10-07 17 11 29](https://cloud.githubusercontent.com/assets/11281373/10351438/8b8a312a-6d17-11e5-8a9e-62beef50393c.png)

**Comment by jamesmcclain** [comment]
Update: this is happening on my local setup but not on mmw.azavea.com, which should be identical at this moment, so the problem might be mine.

**Comment by caseypt** [comment]
I got this locally and on mmw.azavea.com. I think it's a problem with the caching.

Here's the stack trace from Celery:

```
[2015-10-08 11:09:03,619: ERROR/MainProcess] Task apps.modeling.tasks.run_tr55[bb18452e-1804-495e-9978-640f470a178f] raised unexpected: Exception(u'Missing pieces or censuses for modifications',)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/app/apps/modeling/tasks.py", line 185, in run_tr55
    raise Exception('Missing pieces or censuses for modifications')
Exception: Missing pieces or censuses for modifications
[2015-10-08 11:09:03,627: ERROR/Worker-1] Task bb18452e-1804-495e-9978-640f470a178f run from job 1339 raised exception: Missing pieces or censuses for modifications
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/app/apps/modeling/tasks.py", line 185, in run_tr55
    raise Exception('Missing pieces or censuses for modifications')
Exception: Missing pieces or censuses for modifications
```

The modification census is being sent, but I think it's out of date, so this may be a problem with this condition: https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/apps/modeling/views.py#L255.

I'm a little confused because this is pretty simple to trigger, but I didn't encounter it originally. I'll keep on investigating.

**Comment by caseypt** [comment]
The problem is here: https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/apps/modeling/tasks.py#L183-L185

The conditional is a little too aggressive. It's a problem if we have modification pieces but no censuses, however, we can have a census but no pieces. In the case described above, a census was generated for the scenario when the forest was added. When the forest is deleted, we run tr-55. We POST the census from last time, but we correctly identify that we don't need to run any geoprocessing task because we already have the AoI census, and we have no modification pieces so we don't need a modification census. However, when we get to the above mentioned line we throw an exception.

I did this to prevent problems with https://github.com/WikiWatershed/model-my-watershed/blob/develop/src/mmw/apps/modeling/tasks.py#L187-L188. But, if there are no pieces, we should clear out the modification censuses. Then when `build_tr55_modification_input` is run, it will return an empty result.

Working on a fix.
